analysis_driver/report_generation/run_crawler.py

```diff
@@ -217,9 +217,17 @@ def _populate_barcode_info_from_json_file(self, run_dir):
         barcode_info variable. It succeeds and will replace the _populate_barcode_info_from_conversion_file() and
         _populate_barcode_info_from_adapter_file() functions."""
         json_files = util.find_files(run_dir, 'Stats', 'Stats.json')
-        if json_files:
-            with open(json_files[0], 'r') as json_stats:
-                json_data = json.load(json_stats)
+        if not json_files:
+            json_files = util.find_files(run_dir, 'Stats', 'lane_*_Stats.json')
+
+        if not json_files:
+            # Either the full or split file are expected.
+            # The process should stop if they are not found.
+            raise FileNotFoundError()
+
+        for json_file in json_files:
+            with open(json_file, 'r') as open_file:
+                json_data = json.load(open_file)
```

Comment:
Here we are iterating over different JSON files split by lane, and then running code that populates info based on run element ID. Was wondering if we could end up overwriting data in `self.barcodes_info`, but since run element ID depends on the lane number, I think this should be fine?

Reply:
If the json files had metrics for overlapping lanes (lane 1 and 2 in first one then lane 2 in the second one) then that could be a problem but it should not happen.
In any case the metrics could be overwritten but not the `self.barcodes_info` because this is only defined in `_populate_barcode_info_from_dataset`